app/decorators/identity_decorator.rb

```diff
@@ -1,9 +1,32 @@
 IdentityDecorator = Struct.new(:identity) do
+  include ActionView::Helpers::TagHelper
+
   def pretty_event_type
-    I18n.t('event_types.authenticated_at', service_provider: identity.display_name)
+    service_provider_html = if return_to_sp_url.present?
+                              return_to_sp_link
+                            else
+                              identity.display_name
+                            end
+    I18n.t('event_types.authenticated_at', service_provider: service_provider_html).html_safe
```

Comment:
I _think_ it is okay to call `#html_safe` here since all the parts of this string are hardcoded in our `service_providers.yml` file and is not input by users.

Reply:
Would it make sense to at minimum have some validation that these are not `javascript://` URLs? It's easy to imagine accidentally not having appropriate validation on ingestion of URLs, or changing the behavior in the future and not realizing the consequences.

Reply:
`html_safe` is such a bad code smell for something like this -- could we use `link_to` please? Or maybe break out a partial for the different types?

Reply:
And if we expect html stuff inside translations we should make sure the i18n key ends in `_html` so that the i18n gem will auto-html-safe for us

Reply:
Ok I just looked at what this is doing -- we should rename the key to have HTML and probably have separate keys for HTML and non-HTML (so we dont accidentlaly allow a bad SP name to inject HTML into our page)

Reply:
I see about using `link_to`. I didn't because including the helper in a decorator means pulling in the application routing context, but that can be fixed by moving this into a partial.

Reply:
`link_to` can take full URLs as arguments too? I think it will be ok?